com.unity.netcode.gameobjects/Editor/CodeGen/NetworkBehaviourILPP.cs

```diff
@@ -669,7 +669,7 @@ private bool GetWriteMethodForParameter(TypeReference paramType, out MethodRefer
                     {
                         if (parameters[1].IsIn)
                         {
-                            if (parameters[1].ParameterType.Resolve() == paramType.MakeByReferenceType().Resolve() &&
+                            if (((ByReferenceType)parameters[1].ParameterType).ElementType.FullName == paramType.FullName &&
                                 ((ByReferenceType)parameters[1].ParameterType).ElementType.IsArray == paramType.IsArray)
```

Comment:
```suggestion
                            if (((ByReferenceType)parameters[1].ParameterType).ElementType.FullName == paramType.FullName)
```
Presumably you don't need the array check anymore since fullname should include `[]`

Reply:
I THINK you're right on that, but I'm not confident enough in that thought to remove it. My thought was that leaving it wouldn't hurt anything, but I can't completely guarantee removing it won't... I think we have tests that should cover it if that breaks but again, not 100% sure that they exist or are thorough enough. I personally just kind of feel risk-averse in this part of the codebase.